katcp/testutils.py

```diff
@@ -699,19 +701,28 @@ def test_help(self, expected_requests, full_descriptions=False,
 
         if descriptions:
             if not full_descriptions:
-                got_requests = [(name, desc.split('\n')[0])
+                got_requests = [(name, desc.split(b'\n')[0])
                                 for (name, desc) in got_requests]
-                expected_requests = [(name, desc.split('\n')[0])
-                                     for (name, desc) in expected_requests]
+                expected_requests = [
+                    (native_str_to_bytes(name),
+                     native_str_to_bytes(desc).split(b"\n")[0])
+                    for (name, desc) in expected_requests
+                ]
         else:
             got_requests = [name for (name, desc) in got_requests]
+            expected_requests = [native_str_to_bytes(name) for name in expected_requests]
+
 
         got_set = set(got_requests)
         expected_set = set(expected_requests)
 
         if exclude_defaults:
             default_request_names = set(DeviceServer._request_handlers.keys())
+            bytes_str_default_request_names = set()
+            for name in default_request_names:
+                bytes_str_default_request_names.add(native_str_to_bytes(name))
 
+            default_request_names = bytes_str_default_request_names
```

Comment:
How about doing all this using a set comprehension instead?
```python
default_request_names = {native_str_to_bytes(name) for name in DeviceServer._request_handlers.keys()}
```

Or using map:
```python
default_request_names = set(map(native_str_to_bytes, DeviceServer._request_handlers.keys())
```

(Untested)

Reply:
I wanted to be a bit explicit for future persons debugging this as it is test code and not production code but I will add the set comprehension. 
I also noticed that the tests have been failing since this change so I have to look into it.

Reply:
Thanks.  I think that Python 3.8 failure is a flaky test.  I ran it again, and it passed.  Now coverage is indicating a problem?